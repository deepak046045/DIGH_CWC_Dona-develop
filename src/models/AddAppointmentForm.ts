import { AddAppointmentFormFields } from '@/shared/enums/AddAppointmentFormFields';
import { AppointmentTypes } from '@/shared/enums/AppointmentTypes';

export type AddAppointmentForm = {
  [AddAppointmentFormFields.APPOINTMENT_TYPE]: AppointmentTypes | '';
  [AddAppointmentFormFields.INFUSION_NUMBER]: string;
  [AddAppointmentFormFields.ADDRESS]: string;
  [AddAppointmentFormFields.SCHEDULE_DATE]: string;
  [AddAppointmentFormFields.SCHEDULE_TIME]: number;
  [AddAppointmentFormFields.CARE_PARTNERS]: string[];
};
