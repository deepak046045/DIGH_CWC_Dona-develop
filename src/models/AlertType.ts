export type AlertType = {
  isOpen: boolean;
  message: string;
  level: string;
};
