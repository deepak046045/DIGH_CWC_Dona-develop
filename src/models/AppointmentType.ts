import { AppointmentTypes } from '@/shared/enums/AppointmentTypes';

export type AppointmentType = {
  name: AppointmentTypes;
  count?: number;
};

export const APPOINTMENT_ICONS: { [key in AppointmentTypes]: string } = {
  [AppointmentTypes.INFUSION]: '/icons/iv-infusion-icon.svg',
  [AppointmentTypes.GENERAL_DOCTOR]:
    '/icons/general-practicioner-check-in-icon.svg',
  [AppointmentTypes.NEUROLOGIST]: '/icons/neurologist-icon.svg',
  [AppointmentTypes.SAFETY_MRI]: '/icons/safety-mri-icon.svg',
};
