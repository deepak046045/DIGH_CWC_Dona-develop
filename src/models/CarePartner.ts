export type CarePartner = {
  id: string;
  name: string;
  email: string;
};
