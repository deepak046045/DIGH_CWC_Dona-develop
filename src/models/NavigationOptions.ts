export type NavigationOption = {
  title: string;
  path: string;
};

export const NavigationOptions: Record<string, NavigationOption> = {
  TreatmentMap: {
    title: 'Treatment map',
    path: '/treatmentMap',
  },
  WhatToExpect: {
    title: 'What to expect',
    path: '/whatToExpect',
  },
  MyTreatmentPlan: {
    title: 'My treatment plan',
    path: '/myTreatmentPlan',
  },
  MyAppointments: {
    title: 'My appointments',
    path: '/myAppointments',
  },
  AddAppointment: {
    title: 'Add appointment',
    path: '/addAppointment',
  },
  MyCarePartners: {
    title: 'My care partners',
    path: '/myCarePartners',
  },
  MyProfile: {
    title: 'My profile',
    path: '/myProfile',
  },
};

export const gatedNavigationOptions: NavigationOption[] = [
  NavigationOptions.MyTreatmentPlan,
  NavigationOptions.MyAppointments,
  NavigationOptions.MyCarePartners,
  NavigationOptions.WhatToExpect,
];

export const ungatedNavigationOptions: NavigationOption[] = [
  NavigationOptions.TreatmentMap,
  NavigationOptions.WhatToExpect,
];
