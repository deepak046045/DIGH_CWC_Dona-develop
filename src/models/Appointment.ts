import { AppointmentType } from './AppointmentType';
import { CarePartner } from './CarePartner';

export type Appointment = {
  id: string;
  date: Date;
  address: string;
  type: AppointmentType;
  carePartners: CarePartner[];
};
