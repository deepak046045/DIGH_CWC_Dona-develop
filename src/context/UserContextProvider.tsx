'use client';

import React, {
  Dispatch,
  ReactNode,
  SetStateAction,
  createContext,
  useMemo,
  useState,
} from 'react';
import {
  NavigationOption,
  ungatedNavigationOptions,
} from '@/models/NavigationOptions';
import { Appointment } from '@/models/Appointment';
import { AlertType } from '@/models/AlertType';
import mockAppointments from '@/mocks/mockAppointments';
import { CarePartner } from '@/models/CarePartner';
import { mockCarePartners } from '@/mocks/mockCarePartners';

type UserContextDataType = {
  isLoggedIn: boolean;
  setIsLoggedIn: Dispatch<SetStateAction<boolean>>;
  activeView: NavigationOption;
  setActiveView: Dispatch<NavigationOption>;
  alert: AlertType;
  setAlert: Dispatch<SetStateAction<AlertType>>;
  appointments: Appointment[];
  setAppointments: Dispatch<SetStateAction<Appointment[]>>;
  carePartners: CarePartner[];
  setCarePartners: Dispatch<SetStateAction<CarePartner[]>>;
};

export const UserContext = createContext<UserContextDataType>({
  isLoggedIn: false,
  setIsLoggedIn: () => {},
  activeView: ungatedNavigationOptions[0],
  setActiveView: () => {},
  alert: { isOpen: false, message: '', level: '' },
  setAlert: () => {},
  appointments: mockAppointments,
  setAppointments: () => {},
  carePartners: mockCarePartners,
  setCarePartners: () => {},
});

type Props = {
  children: ReactNode;
};

export function UserContextProvider({ children }: Props) {
  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
  const [activeView, setActiveView] = useState<NavigationOption>(
    ungatedNavigationOptions[0]
  );
  const [alert, setAlert] = useState<AlertType>({
    isOpen: false,
    message: '',
    level: '',
  });
  const [appointments, setAppointments] =
    useState<Appointment[]>(mockAppointments);
  const [carePartners, setCarePartners] =
    useState<CarePartner[]>(mockCarePartners);

  const values = useMemo(
    () => ({
      isLoggedIn,
      setIsLoggedIn,
      activeView,
      setActiveView,
      alert,
      setAlert,
      appointments,
      setAppointments,
      carePartners,
      setCarePartners,
    }),
    [isLoggedIn, activeView, alert, appointments, carePartners]
  );

  return <UserContext.Provider value={values}>{children}</UserContext.Provider>;
}
