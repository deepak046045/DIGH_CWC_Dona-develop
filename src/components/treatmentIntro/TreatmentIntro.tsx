'use client';

import React, { useContext } from 'react';
import { LdsButton } from '@elilillyco/ux-lds-react';
import { UserContext } from '@/context/UserContextProvider';
import { copyTexts } from '@/constants/copyTexts';
import layout from '@/scss/themes/layout.module.scss';
import { useRouter } from 'next/navigation';
import { NavigationOptions } from '@/models/NavigationOptions';
import treatmentIntro from './treatmentIntro.module.scss';

export default function TreatmentIntro() {
  const router = useRouter();
  const { isLoggedIn, setActiveView } = useContext(UserContext);

  return (
    <div className={`${treatmentIntro.TreatmentIntro} ${layout.gutter}`}>
      <h2 className={treatmentIntro.TreatmentIntro__title}>
        {isLoggedIn
          ? copyTexts.TreatmentIntro.titleGated
          : copyTexts.TreatmentIntro.titleUngated}
      </h2>
      <p className={treatmentIntro.TreatmentIntro__subtitle}>
        {copyTexts.TreatmentIntro.subtitleUngated}
      </p>
      <div className={treatmentIntro.TreatmentIntro__wrapper}>
        <p className={treatmentIntro.TreatmentIntro__text}>
          {copyTexts.TreatmentIntro.textUngated}
        </p>
        {isLoggedIn ? (
          <LdsButton
            data-testid="add-appointment-button"
            classes={`${treatmentIntro['TreatmentIntro__appointment-btn']}`}
            onClick={() => {
              router.push(NavigationOptions.AddAppointment.path);
              setActiveView(NavigationOptions.MyAppointments);
            }}
          >
            {copyTexts.TreatmentIntro.addApointmentBtnGated}
          </LdsButton>
        ) : null}
      </div>
    </div>
  );
}
