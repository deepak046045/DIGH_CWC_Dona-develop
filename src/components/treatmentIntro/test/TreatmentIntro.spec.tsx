import React from 'react';
import { UserContext } from '@/context/UserContextProvider';
import { fireEvent, render, screen } from '@testing-library/react';
import TreatmentIntro from '../TreatmentIntro';
import { useRouter } from 'next/navigation';
import { NavigationOptions } from '@/models/NavigationOptions';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(() => ({ push: jest.fn() })),
}));

describe('TreatmentIntro', () => {
  it('should render page when logged in', () => {
    const mockValue = {
      isLoggedIn: true,
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <TreatmentIntro />
      </UserContext.Provider>
    );

    expect(screen.queryByTestId('add-appointment-button')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the page when not logged in', () => {
    const mockValue = {
      isLoggedIn: false,
    };

    const { container } = render(
      <UserContext.Provider value={mockValue}>
        <TreatmentIntro />
      </UserContext.Provider>
    );
    expect(
      screen.queryByTestId('add-appointment-button')
    ).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render page when logged in', () => {
    const setActiveViewMock = jest.fn();
    const pushMock = jest.fn();

    const mockValue = {
      isLoggedIn: true,
      setActiveView: setActiveViewMock,
    };
    (useRouter as jest.Mock).mockReturnValue({
      query: {},
      push: pushMock,
    });

    render(
      <UserContext.Provider value={mockValue}>
        <TreatmentIntro />
      </UserContext.Provider>
    );

    const addAppointmentButton = screen.getByTestId('add-appointment-button');
    fireEvent.click(addAppointmentButton);

    expect(pushMock).toHaveBeenCalledWith(
      NavigationOptions.AddAppointment.path
    );
    expect(setActiveViewMock).toHaveBeenCalledWith(
      NavigationOptions.MyAppointments
    );
  });
});
