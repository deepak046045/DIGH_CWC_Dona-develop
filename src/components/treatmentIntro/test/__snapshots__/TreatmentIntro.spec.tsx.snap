// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreatmentIntro should render page when logged in 1`] = `
<div>
  <div
    class="TreatmentIntro gutter"
  >
    <h2
      class="TreatmentIntro__title"
    >
      My treatment plan
    </h2>
    <p
      class="TreatmentIntro__subtitle"
    >
      Setting you up for success in your treatment. Setting you up for success in your treatment.
    </p>
    <div
      class="TreatmentIntro__wrapper"
    >
      <p
        class="TreatmentIntro__text"
      >
        This treatment map serves as a suggested guide. Always follow your doctor’s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.
      </p>
      <button
        class="lds-button TreatmentIntro__appointment-btn"
        data-testid="add-appointment-button"
        type="button"
      >
        Add Appointment
      </button>
    </div>
  </div>
</div>
`;

exports[`TreatmentIntro should render the page when not logged in 1`] = `
<div>
  <div
    class="TreatmentIntro gutter"
  >
    <h2
      class="TreatmentIntro__title"
    >
      Recommended treatment
    </h2>
    <p
      class="TreatmentIntro__subtitle"
    >
      Setting you up for success in your treatment. Setting you up for success in your treatment.
    </p>
    <div
      class="TreatmentIntro__wrapper"
    >
      <p
        class="TreatmentIntro__text"
      >
        This treatment map serves as a suggested guide. Always follow your doctor’s directions. Insurance approval may be needed or the start of your infusion therapy may be delayed.
      </p>
    </div>
  </div>
</div>
`;
